source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -197,6 +171,38 @@ FilterConfig::FilterConfig(const envoy::extensions::filters::http::ext_authz::v3
   }
 }
 
+// Initialize the appropriate cache, out of two possible caches, based on configuration.
+void FilterConfig::initializeCache(Server::Configuration::ServerFactoryContext& factory_context) {
+  // If response_cache_config section doesn't exist in the config, no need to create cache objects.
+  if (!response_cache_config_.has_value()) {
+    response_cache_status_ = nullptr;
+    response_cache_full_ = nullptr;
+    return;
+  }
+
+  if (response_cache_remember_body_headers_) {
+    // Initialize the cache that stores full response (body, header modification, metadata, status).
+    auto cache_result = Envoy::Common::RespCache::createCache<Filters::Common::ExtAuthz::Response>(
```

Comment:
With an extension config, you'll declare a factory interface, and then call something like 

```
Config::Utility::getAndCheckFactory<ExtAuthzFactory>(
          config)
```

And each extension will register it's factory with that type.